### PR TITLE
Improve converter testing

### PR DESCRIPTION
**Pull request by lampajr**
## Description

Improve unit testing for the `converter` package.

## How Has This Been Tested?

```bash
make test
```

## TODOs
- [x] Merge https://github.com/opendatahub-io/model-registry/pull/140
- [x] Rebase from `main` once https://github.com/opendatahub-io/model-registry/pull/140 got merged

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: internal/converter/mlmd_converter_util_test.go [hunk 1]
```diff
@@ -44,6 +44,25 @@ func TestInt64ToString(t *testing.T) {
...
 }
 
+func TestStringToInt32(t *testing.T) {
+	assertion := setup(t)
+
+	valid := "12345"
+	converted, err := StringToInt32(valid)
+	assertion.Nil(err)
+	assertion.Equal(int32(12345), converted)
+	assertion.Nil(StringToInt64(nil))
```
Comment (tarilabs) [inline internal/converter/mlmd_converter_util_test.go]: not sure about this line, seems is using the int64 variant... but not sure what was the intent

**Review by tarilabs — DISMISSED** [review]
minor comment, looks good to me, thanks!

File: internal/converter/mlmd_converter_util_test.go (hunk 1 above)
Comment (lampajr) [inline internal/converter/mlmd_converter_util_test.go]:
```suggestion
	assertion.Nil(StringToInt32(nil))
```

Comment (lampajr) [reply internal/converter/mlmd_converter_util_test.go]: it was indeed a leftover, fixed :rocket:

File: internal/converter/mlmd_converter_util_test.go [hunk 2]
```diff
@@ -44,6 +44,24 @@ func TestInt64ToString(t *testing.T) {
...
+	assertion.Equal(int32(12345), converted)
+}
+
+func TestStringToInt32InvalidNumber(t *testing.T) {
+	assertion := setup(t)
+
+	invalid := "not-a-number"
+	converted, err := StringToInt32(invalid)
+	assertion.NotNil(err)
+	assertion.Equal(int32(0), converted)
```
Comment (tarilabs) [inline internal/converter/mlmd_converter_util_test.go]:
```suggestion

```

I'd also remove this line, as the Tuple_0 when error in Tuple_1 is present should be "unspecified" my2c

**Review by tarilabs — DISMISSED** [review]

File: internal/converter/mlmd_converter_util_test.go (hunk 2 above)
Comment (lampajr) [reply internal/converter/mlmd_converter_util_test.go]: make sense!

Comment (lampajr) [reply internal/converter/mlmd_converter_util_test.go]: had to ignore the function result using `_` :)

**Review by tarilabs — APPROVED** [review]
Thank you, sorry for spotting on the second round only

**Pull request merged**